### PR TITLE
docs(repo): add missing rules

### LINKED ISSUE #2193: Add missing rules to spectral docs
**Chore summary**
Add missing rules to spectral [docs](https://meta.stoplight.io/docs/spectral/4dec24461f3af-open-api-rules#openapi-tags-uniqueness):

- oas2-valid-media-example
- oas3-valid-media-example
- oas3-valid-schema-example

### PR DESCRIPTION
**Pull request by heitortsergent**
Fixes #2193.

Adds the following 3 rules that were missing from the OpenAPI Rules doc:

- oas2-valid-media-example
- oas3-valid-media-example
- oas3-valid-schema-example

**Checklist**

- [ ] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by heitortsergent** [comment]
Questions I have for this PR:

- Are the examples I have added for the rules correct/ok?
- Are the 3 rules **Recommended** supposed to be "yes"? I marked them as "yes" by looking at [rulesets/src/oas/index.ts](https://github.com/stoplightio/spectral/blob/develop/packages/rulesets/src/oas/index.ts#L659).
- Can I reorder the rules in the OpenAPI Rules page so they're in alphabetical order?

**Review by philsturgeon — DISMISSED** [review]

**Comment by philsturgeon** [comment]
Yes, yes, and yes!

File: docs/reference/openapi-rules.md [hunk 1]
```diff
@@ -658,3 +727,122 @@ Validate structure of OpenAPI v3 specification.
 Parameter objects should have a `description`.
 
 **Recommended:** No
+
+### oas3-valid-media-example
+
+Examples must be valid against their defined schema. This rule is applied to Media objects, such as requests and responses.
```
Comment (P0lip) [inline docs/reference/openapi-rules.md]: request and responses aren't media objects though.
Media Type Object is just a standalone object https://spec.openapis.org/oas/v3.1.0#media-type-object that may be present under "content" in requests or responses

**Review by P0lip — CHANGES_REQUESTED** [review]
Looks good apart from that one comment

**Comment by heitortsergent** [comment]
@P0lip thanks for the review! I updated the description and removed the request/response mention.

Any thoughts on if I can reorder the rules alphabetically? 🤓

File: docs/reference/openapi-rules.md (hunk 1 above)
Comment (philsturgeon) [reply docs/reference/openapi-rules.md]: I think this is correct in the sense that a media type object is used in a request and a response. Requests and responses both have media types in there somewhere.

**Comment by philsturgeon** [comment]
> Any thoughts on if I can reorder the rules alphabetically? 🤓 

Absolutely.

**Review by pamgoodrich — APPROVED** [review]

**Review by P0lip — APPROVED** [review]

**Pull request merged**